### ISSUE TITLE
HTML frontend to tumbler.py

### ISSUE THREAD

**Issue by chris-belcher** [chris-belcher/joinmarket#108]
Something to improve user friendliness of tumbler.py, have a HTML page a bit like ob-watcher.py which walks the user through starting up a tumbler. Essentially making the process about as easy as BitcoinFog or similar centralized mixers.

The page could tell the user to deposit funds to 2 or 3 addresses and allow the user to input some destination addresses, along the way explaining the benefits of using several addresses for input and output as outlined in #28 . It would then print out useful information telling the user how the tumbling is going.

There could be QR codes shown.

**Comment by chris-belcher** [comment]
Although HTML is easy, it would also make it easier for someone to run a web server and allow people to mix their coins via a third party. This is not something to be encouraged so gui-tumbler will have to be an actual GUI with PyQt or something.

**Comment by chris-belcher** [comment]
![gui-tumbler-pic](https://cloud.githubusercontent.com/assets/8398185/15093049/5fe3f50c-1473-11e6-88a5-4bc318d22600.png)


My drawing of the what GUI would look like.

It's styled along the lines of an install wizard that walks the user through what they have to do.

It creates a wallet by default, accepts coins and mixes, there is red button flashing until the user clicks it and writes down the 12 words.
All the defaults are sane for privacy, if a user does something stupid it will tell them.

**Comment by chris-belcher** [comment]
tumbler should have #467 done first i think

**Comment by alvinjoelsantos** [comment]
being worked on.. see:  https://github.com/coinmix/coinmix

**Comment by chris-belcher** [comment]
@alvinjoelsantos It looks like you've taken all the ideas written here and then swapped out the donation address with your own.

**Comment by alvinjoelsantos** [comment]
@chris-belcher I took the view that it is more efficient to merge all the outstanding issues together spinning out a separate project with the sole purpose of developing a user-friendly client of tumbler that is packaged into a .deb file. The project is operating under GPL and anyone is free to use, copy, or modify the code. Any improvements made to the tumbler code can be committed back to joinmarket. Isn't that whole point of forking? As for donation address issue, the idea is to have a budget for marketing, expenses, and instituting a payment incentive policy for contributors per commit to encourage the develop of a polished application. The potential of increasing the user base benefits everyone, especially yield-generator users.

**Comment by chris-belcher** [comment]
At a time when JoinMarket is [under attack](https://github.com/JoinMarket-Org/joinmarket/issues/156#issuecomment-224433684) and has [sybil problems coming to light that need fixing](https://www.reddit.com/r/joinmarket/comments/4mgl6w/join_me_on_a_market_for_anonymity_32k_capital_to/), it's somewhat disheartening that out of all the issues, this is the one that gets worked on.

There's a ton of issues marked [easy to implement](https://github.com/JoinMarket-Org/joinmarket/issues?q=is%3Aopen+is%3Aissue+label%3A%22Easy+%2F+Quick+to+Implement%22) which get very little attention, even highly sought-for features like the Electrum plugin [cant seem to find testers](https://bitcointalk.org/index.php?topic=919116.msg15148759#msg15148759). *I've gotten only 1 person so far after more than a week who even installed it, which is disappointing after *so* many requests in the early days to do an Electrum plugin.*

Yet in all this, a new account appears as soon as there's money to be made (or as it's called, a budget for marketing, expenses and paying committers)

**Comment by alvinjoelsantos** [comment]
Yes, there are there's a ton of issues as you say, but you have to start somewhere. What is disheartening is that when someone takes the initiative to tackle a number of issues and consolidate them into a separate project, you get this kind of reaction. In regards to have a budget, this issue has been brought up before [here](https://github.com/JoinMarket-Org/joinmarket/issues/258)  A development fund is needed to have a sustainable project. No one expects to make "money" out of this. A person's time and efforts would be best used in other things if this was the main motivation. I do understand where you are coming from though that there should be a focus in dealing with core fundamental issues that might call in to question the whole infrastructure itself. You have to put out the fire in the house, instead of rearranging the furniture. But you guys are in the thick of it and know the system a lot better. I will try to help any way I can.
